server/controllers/traineeController.js

```diff
@@ -1,11 +1,55 @@
+import db from "../db";
+
 export const getAllTrainees = (req, res) => {
     res.status(200).json("hello from trainees route");
 };
 
-export const getTrainee = (req, res) => {
-    res.status(200).json("you can write your get function to get one trainee here");
+export const getTrainee = async (req, res) => {
+    const { traineeId } = req.params;
+
+    try {
+        const q = "SELECT * FROM trainees WHERE id = $1";
+        const result = await db.query(q, [traineeId]);
+        const trainee = result.rows[0];
+
+        if (!trainee) {
+            return res.status(404).json({ error: "Trainee not found" });
+        }
+
+        res.status(200).json({ trainee });
+    } catch (error) {
+        console.error("Error retrieving trainee:", error);
+        res.status(500).send("Failed, Internal Server Error. Database or table not found");
```

Comment:
Probably no need to tell the API user 'Database or table not' as there isnt anything they can do about it

Reply:
thank you, i just copied so with mistake : ), hev ebeen removed